src/Generators/ModelGenerator.php

```diff
@@ -145,9 +145,18 @@ private function buildRelationships(Model $model)
                 }
 
                 $name = Str::beforeLast($name, '_id');
-                $class = Str::studly($class ?? $name);
-                $relationship = sprintf("\$this->%s(%s::class)", $type, '\\' . $model->fullyQualifiedNamespace() . '\\' . $class);
-
+                $relationShipClass = Str::studly($class ?? $name);
+
+                $relationship = "";
+                if (!is_null($class) && $type === 'belongsToMany') {
+                    if (preg_match("/^[A-Z]/", $name)) {
```

Comment:
It's not clear that this `preg_match` determines if the developer passed a _model reference_.

I am open to supporting this. But I feel there are more implications (e.g. generating the pivot model). For now, I think we should focus on the original issue, which was specifying the custom table name.

Said another way, let's assume, for now, that whatever they pass is a custom table name.